paywall/src/middlewares/web3Middleware.js

```diff
@@ -116,12 +120,31 @@ export default function web3Middleware({ getState, dispatch }) {
         }
       }
 
-      if (action.type === ADD_TRANSACTION) {
-        web3Service.getTransaction(action.transaction.hash)
-      }
-
-      if (action.type === NEW_TRANSACTION) {
-        web3Service.getTransaction(action.transaction.hash, action.transaction)
+      if (action.type === ADD_TRANSACTION || action.type === NEW_TRANSACTION) {
+        const defaults =
+          action.type === NEW_TRANSACTION ? action.transaction : undefined
+        web3Service.getTransaction(action.transaction.hash, defaults)
+        const { hash } = action
+        pollWithConditions(
+          // the polling function simply returns the transaction with the closed over transaction hash
+          // note that a unique poll starts for each NEW_TRANSACTION call
+          () => {
+            web3Service.getTransaction(hash)
```

Comment:
We certainly would want to keep the `defaults` value here because nothing guarantees that on the second call to `web3Service.getTransaction` we would have seen a submitted transaction...